Endeavour Health - admin and appointment resources, extensions and valuesets

jonnyry opened the pull request:
Changes to all resources

```
* Change comment of id to request globally unique identifier where element is present
* Fix and require profile URL in meta.profile slice
* Change resource references to reference CareConnect profile versions  
```

Create new valuesets

```
* EthnicCategory valueset using values from NHS Data Dictionary
* HumanLanguage valueset using values from NHS Data Dictionary
* OrganizationType valueset using values from NHS Data Dictionary
* NhsNumberVerificationStatus valueset using values from NHS Data Dictionary
```

Create new extensions

```
* ActivePeriod extension for Organization and Location resources
* AdditionalActor extension for Schedule resource
* EthnicCategory extension for Patient resource
* MainLocation extension for Organization resource
* Location extension for Schedule resource
* NhsNumberVerificationStatus extension for Patient resource
* AppointmentArrived extension for Appointment resource
* AppointmentSentIn extension for Appointment resource
* AppointmentLeft extension for Appointment resource
```

Changes to Organization resource

```
* Add odsCode slice to identifier
* Add localIdentifier slice to identifier
* Change cardinality of name to 1..1
* Remove address.state
* Change type binding to OrganizationType valueset
* Change contact.name.family to 1..1
* Change contact.name.prefix and contact.name.suffix to 0..1
* Remove contact.name.text, contact.name.use and contact.name.period
* Add activePeriod extension to active
* Add mainLocation extension
```

Changes to Location resource

```
* Change cardinality of name to 1..1
* Remove address.state
* Add activePeriod extension to status
```

Changes to Patient resource

```
* Add nhsNumber slice to identifier
* Add nhsNumberVerificationStatus extension to nhsNumber slice
* Add chiNumber slice to identifier
* Add hcNumber slice to identifier
* Change cardinality of name to 1..1
* Change name.family and contact.name.family to 1..1
* Change name.prefix, name.suffix, contact.name.prefix and contact.name.suffix to 0..1
* Remove name.text, name.use, name.period, contact.name.text, contact.name.use and contact.name.period
* Remove address.state and contact.address.state
* Remove animal
* Add ethnicCategory extension
* Change communication.language binding to HumanLanguage valueset
```

Changes to Practitioner resource

```
* Add sdsUserId slice to identifier
* Add sdsRoleProfileId slice to identifier
* Change cardinality of name to 1..1
* Change name.family to 1..1
* Change name.prefix and contact.name.suffix to 0..1
* Remove name.text, name.use and name.period
* Remove address.state
* Change communication.language binding to HumanLanguage valueset
```

Changes to Schedule resource

```
* Restrict actor to Practitioner
* Change planningHorizon to 1..1
* Change planningHorizon.start to 1..1
* Add additionalActors extension
* Add location extension    
```

Changes to Slot resource

```
* Change schedule.reference to 1..1
* Remove schedule.display
```

Changes to Appointment resource

```
* Change slot to 1..*
* Change slot.reference to 1..1
* Remove slot.display
* Add arrived extension
* Add sentIn extension
* Add left extension
```

Commit:
Organization-1 and associated resources

Organization-1:

* Changed comment of id element to require population with globally
unique identifier
* Added odsCode slice to identifier element with 0..1 cardinality
* Added localIdentifier slice to identifier element with 0..*
cardinality
* Changed Organization references to use the profile URL
* Added ActivePeriod-1 extension to active element
* Bound type element to OrganizationType-1 valueset
* Made name element mandatory

ActivePeriod-1
* Create extension to record open and close dates for an organization

OrganizationType-1
* Create organisation type valueset from NHS data dictionary at
http://www.datadictionary.nhs.uk/data_dictionary/attributes/o/org/organi
sation_type_de.asp?shownav=0

Commit:
Updating naming for extensions

Commit:
Organization-1, Location-1 and associated resources

Organization-1:

* Added mainLocation extension

Location-1:

* Changed comment of id element to require population with globally
unique identifier
* Made system and value mandatory in identifier element
* Changed Location and Organization references to use the profile URLs
* Added ActivePeriod-1 extension to statu element
* Made name element mandatory
* Removed address.state

MainLocation-1
* Extension to carry a reference to the main location for an
organization

ActivePeriod-1
* Reverted previous naming change

Commit:
Schedule-1, Slot-1, Appointment-1

Schedule-1

* Add base CareConnect resource

Slot-1

* Add base CareConnect resource

Appointment-1

* Add base CareConnect resource

Commit:
Organization-1, OrganizationType-1

Fix valueset code system in both

Commit:
Schedule-1 and associated resources

Schedule-1

* Change description for id element to require globally unique
identifier where populated
* Mandate identifier.system and identifier.value where identifier is
populated
* Restrict actor to be a Practitioner-1 and make mandatory
* Require planningHorizon and planningHorizon.start
* Add additionalActors extension to allow further practitioners to be
added to the schedule
* Add location extension

AdditionalActor-1-Extension

* Add to support Schedule-1

Location-1-Extension

* Add to support Schedule-1

Commit:
Slot-1

Slot-1

* Change description for id element to require globally unique
identifier where populated
* Mandate identifier.system and identifier.value where identifier is
populated
* Require a schedule reference to be populated
* Change the schedule reference to reference a Schedule-1

Commit:
Patient-1

Patient-1

* Add UK national identifier slices - NHS number, CHI number, H&C number
* Change description on id element to require a globally unique
identifier (where present)
* Remove address.state and address.period
* Remove animal element
* Add ethnicCategory extension
* Change references to other resources to point to CareConnect versions

EthnicCategory-1 (valueset and extension)

* Add valueset from NHS Data Dictionary
* Create extension to support valueset

Commit:
Organization-1

Organization-1

* Add OrganizationType-1 valueset binding

Commit:
Schedule-1

Schedule-1

* Revert identifier element changes

Commit:
Patient-1, HumanLanguage-1

Patient-1

* Change name to 1..1
* Change name.family to 1..1
* Change name.prefix, name.suffix to 0..1
* Remove name.period, name.use
* Bind communication.language to HumanLanguage-1 valueset

HumanLanguage-1

* Create HumanLanguage-1 valueset from NHS Data Dictionary

Commit:
Practitioner-1

Practitioner-1

* Add sdsUserId and sdsRoleProfileId slices to identifier
* Change description of id element to request a globally unique
identifier where populated
* Change cardinality of name to 1..1
* Require a single familyName
* Change prefix and suffix to 0..1
* Remove name.period, telecom.period, address.period
* Remove address.state
* Add HumanLanguage valueset binding to communication

Commit:
All resources

All resources

* Remove 0..0 cardinality restriction on id sub elements for
consistency.  Haven’t restricted all id sub elements, therefore
restriction of sub is inconsistent and not particularly useful

Commit:
Appointment-1

Appointment-1

* Change description of id element to request the use of a globally
unique identifier
* Change cardinality of slot element to 1..*
* Change references to point to CareConnect profiles

Commit:
All resources

All resources

* Require profile URL into meta.profile

Commit:
Organization-1, Location-1, Patient-1, Practitioner-1

Organization-1, Location-1, Patient-1, Practitioner-1

* Revert identifier sub element cardinality changes where no identifier
slices created
* Change cardinalites of name sub elements in line with changes to
previous resources
* Remove address.state
* Revert address and telecom period and use changes

Commit:
Patient-1

* Add new NhsNumberVerification status valueset and extension
* Add nhsNumberVerificationStatus element to nhsNumber identifier slice

Commit:
Appointment-1

* Create three new extensions to capture the patient’s arrived, sent in
and left times for an appointment
* Add the extensions to Appointment-1

Commit:
Condition-1, Problem-1 and associated resources

Condition-1:
* Changed resource references to reference CareConnect profile versions

* Changed code valueset binding strength to Preferred for valueset
reference http://hl7.org/fhir/ValueSet/condition-code
* Made category element mandatory
* Changed category valueset reference to
http://hl7.org.uk/CareConnect-ConditionCategory-1.valueset
* Changed category valueset binding strength to Required
* Made category.coding.system element mandatory with a fixed value of
http://hl7.org.uk/CareConnect-ConditionCategory-1.valueset
* Made category.coding.code element mandatory with a fixed value of
diagnosis
* Made category.coding.display have a fixed value of Diagnosis
* Changed clinicalStatus valueset binding strength to Required for
valueset reference http://hl7.org/fhir/ValueSet/condition-clinical
* Removed string data type from onset[x]
* Removed string data type from abatement[x]
* Removed stage element
* Removed evidence element
* Added problemEpisode element extension -
CareConnect-ConditionProblemEpisode-1-Extension

Problem-1: (New profile based on Condition)
* Changed resource references to reference CareConnect profile versions

* Changed code valueset binding strength to Preferred for valueset
reference http://hl7.org/fhir/ValueSet/condition-code
* Made category element mandatory
* Changed category valueset reference to
http://hl7.org.uk/CareConnect-ConditionCategory-1.valueset
* Changed category valueset binding strength to Required
* Made category.coding.system element mandatory with a fixed value of
http://hl7.org.uk/CareConnect-ConditionCategory-1.valueset
* Made category.coding.code element mandatory with a fixed value of
problem
* Made category.coding.display have a fixed value of Problem
* Made clinicalStatus element mandatory
* Changed clinicalStatus valueset binding strength to Required for
valueset reference http://hl7.org/fhir/ValueSet/condition-clinical
* Removed string data type from onset[x]
* Removed string data type from abatement[x]
* Removed stage element
* Removed evidence element
* Added significance element extension -
CareConnect-ProblemSignificance-1-Extension
* Added expectedDuration element extension -
CareConnect-ProblemExpectedDuration-1-Extension
* Added lastReviewed element extension -
CareConnect-ProblemLastReviewed-1-Extension
* Added related element extension -
CareConnect-ProblemRelationship-1-Extension
* Added associated element extension -
CareConnect-ProblemAssociated-1-Extension

Create new valuesets:
* CareConnect-ConditionCategory-1.valueset
* CareConnect-ConditionEpisodicity-1.valueset
* CareConnect-ProblemRelationshipType-1.valueset
* CareConnect-ProblemSignificance-1.valueset

Create new extensions:
* CareConnect-ConditionProblemEpisode-1-Extension
* CareConnect-ProblemSignificance-1-Extension
* CareConnect-ProblemExpectedDuration-1-Extension
* CareConnect-ProblemLastReviewed-1-Extension
* CareConnect-ProblemRelationship-1-Extension
* CareConnect-ProblemAssociated-1-Extension

Commit:
AllergyIntolerance-1 and associated resources

AllergyIntolerance-1:
* Changed resource references to reference CareConnect profile versions.
* Changed comments on substance element to indicate that a SNOMED
AllergyCode may be sent where a direct substance mapping is not
available in the source system.
* Added qualifier element extension to reaction.manifestation.coding -
CareConnect-CodingQualifier-1-Extension
* Added encounter element extension -
http://hl7.org/fhir/StructureDefinition/encounter-associatedEncounter
* Added endDate element extension -
CareConnect-AllergyIntoleranceEndDate-1-Extension

Create new extensions:
* CareConnect-AllergyIntoleranceEndDate-1-Extension

jonathanhomer commented:
Condition-1, Problem-1 and associated resources

Condition-1:
- Changed resource references to reference CareConnect profile versions
- Changed code valueset binding strength to Preferred for valueset reference http://hl7.org/fhir/ValueSet/condition-code
- Added qualifier element extension to code.coding - CareConnect-CodingQualifier-1-Extension
- Made category element mandatory
- Changed category valueset reference to http://hl7.org.uk/CareConnect-ConditionCategory-1.valueset
- Changed category valueset binding strength to Required
- Made category.coding.system element mandatory with a fixed value of http://hl7.org.uk/CareConnect-ConditionCategory-1.valueset
- Made category.coding.code element mandatory with a fixed value of diagnosis
- Made category.coding.display have a fixed value of Diagnosis
- Changed clinicalStatus valueset binding strength to Required for valueset reference http://hl7.org/fhir/ValueSet/condition-clinical
- Removed string data type from onset[x]
- Removed string data type from abatement[x]
- Removed stage element
- Removed evidence element
- Added problemEpisode element extension - CareConnect-ConditionProblemEpisode-1-Extension 

Problem-1: (New profile based on Condition)
- Changed resource references to reference CareConnect profile versions
- Changed code valueset binding strength to Preferred for valueset reference http://hl7.org/fhir/ValueSet/condition-code
- Added qualifier element extension to code.coding - CareConnect-CodingQualifier-1-Extension
- Made category element mandatory
- Changed category valueset reference to http://hl7.org.uk/CareConnect-ConditionCategory-1.valueset
- Changed category valueset binding strength to Required
- Made category.coding.system element mandatory with a fixed value of http://hl7.org.uk/CareConnect-ConditionCategory-1.valueset
- Made category.coding.code element mandatory with a fixed value of problem
- Made category.coding.display have a fixed value of Problem
- Made clinicalStatus element mandatory
- Changed clinicalStatus valueset binding strength to Required for valueset reference http://hl7.org/fhir/ValueSet/condition-clinical
- Removed string data type from onset[x]
- Removed string data type from abatement[x]
- Removed stage element
- Removed evidence element
- Added significance element extension - CareConnect-ProblemSignificance-1-Extension 
- Added expectedDuration element extension - CareConnect-ProblemExpectedDuration-1-Extension
- Added lastReviewed element extension - CareConnect-ProblemLastReviewed-1-Extension
- Added related element extension - CareConnect-ProblemRelationship-1-Extension
- Added associated element extension - CareConnect-ProblemAssociated-1-Extension

Create new valuesets:
- CareConnect-ConditionCategory-1.valueset
- CareConnect-ConditionEpisodicity-1.valueset
- CareConnect-ProblemRelationshipType-1.valueset
- CareConnect-ProblemSignificance-1.valueset

Create new extensions:
- CareConnect-ConditionProblemEpisode-1-Extension
- CareConnect-ProblemSignificance-1-Extension
- CareConnect-ProblemExpectedDuration-1-Extension
- CareConnect-ProblemLastReviewed-1-Extension
- CareConnect-ProblemRelationship-1-Extension
- CareConnect-ProblemAssociated-1-Extension
- CareConnect-CodingQualifier-1-Extension

AllergyIntolerance-1 and associated resources

AllergyIntolerance-1:
- Changed resource references to reference CareConnect profile versions.
- Changed comments on substance element to indicate that a SNOMED AllergyCode may be sent where a direct substance mapping is not available in the source system.
- Added qualifier element extension to reaction.manifestation.coding - CareConnect-CodingQualifier-1-Extension
- Added encounter element extension - http://hl7.org/fhir/StructureDefinition/encounter-associatedEncounter
- Added endDate element extension - CareConnect-AllergyIntoleranceEndDate-1-Extension 

Create new extensions:
- CareConnect-AllergyIntoleranceEndDate-1-Extension

Commit:
Procedure-1 and associated resources

Procedure-1:
* Changed resource references to reference CareConnect profile versions.
* Restricted Subject to Patient
* Changed code valueset binding strength to Preferred for valueset
reference http://hl7.org/fhir/ValueSet/procedure-code
* Added qualifier element extension to code.coding -
CareConnect-CodingQualifier-1-Extension
* Restricted performer.actor to Organization, Practitioner
* Removed category element
* Removed notPerformed element
* Removed reasonNotPerformed element
* Removed reason[x] element
* Removed outcome element
* Removed report element
* Removed complication element
* Removed followUp element
* Removed request element
* Removed notes.author[x] element
* Removed notes.time element
* Removed focalDevice element
* Removed used element
* Added recordedDate element extension -
CareConnect-RecordedDate-1-Extension

Create new extensions:
* CareConnect-RecordedDate-1-Extension

jonathanhomer commented:
Procedure-1 and associated resources

Procedure-1:
- Changed resource references to reference CareConnect profile versions.
- Restricted Subject to Patient
- Changed code valueset binding strength to Preferred for valueset reference http://hl7.org/fhir/ValueSet/procedure-code
- Added qualifier element extension to code.coding - CareConnect-CodingQualifier-1-Extension
- Restricted performer.actor to Organization, Practitioner
- Removed category element
- Removed notPerformed element
- Removed reasonNotPerformed element
- Removed reason[x] element
- Removed outcome element
- Removed report element
- Removed complication element
- Removed followUp element
- Removed request element
- Removed notes.author[x] element
- Removed notes.time element
- Removed focalDevice element
- Removed used element
- Added recordedDate element extension - CareConnect-RecordedDate-1-Extension 

Create new extensions:
- CareConnect-RecordedDate-1-Extension

Commit:
FamilyMemberHistory-1 and associated resources

FamilyMemberHistory-1:
* Changed resource references to reference CareConnect profile versions.
* Changed relationship valueset binding strength to Preferred for
valueset reference http://hl7.org/fhir/ValueSet/v3-FamilyMember
* Removed string data type from born[x]
* Removed string data type from age[x]
* Removed string data type from deceased[x]
* Removed note.author[x] element
* Removed note.time element
* Made condition element mandatory
* Changed condition.code valueset binding strength to Preferred for
valueset reference http://hl7.org/fhir/ValueSet/condition-code
* Added qualifier element extension to condition.code -
CareConnect-CodingQualifier-1-Extension
* Changed condition.outcome valueset binding strength to Preferred for
valueset reference http://hl7.org/fhir/ValueSet/condition-outcome
* Removed string data type from condition.onset[x]
* Removed condition.note.author[x] element
* Removed condition.note.time element
* Added recorder element extension -
CareConnect-FamilyMemberHistoryRecorder-1-Extension
* Added encounter element extension -
http://hl7.org/fhir/StructureDefinition/encounter-associatedEncounter

Create new extensions:
* CareConnect-FamilyMemberHistoryRecorder-1-Extension

jonathanhomer commented:
FamilyMemberHistory-1 and associated resources

FamilyMemberHistory-1:
- Changed resource references to reference CareConnect profile versions.
- Changed relationship valueset binding strength to Preferred for valueset reference http://hl7.org/fhir/ValueSet/v3-FamilyMember
- Removed string data type from born[x]
- Removed string data type from age[x]
- Removed string data type from deceased[x]
- Removed note.author[x] element
- Removed note.time element
- Made condition element mandatory
- Changed condition.code valueset binding strength to Preferred for valueset reference http://hl7.org/fhir/ValueSet/condition-code
- Added qualifier element extension to condition.code - CareConnect-CodingQualifier-1-Extension
- Changed condition.outcome valueset binding strength to Preferred for valueset reference http://hl7.org/fhir/ValueSet/condition-outcome
- Removed string data type from condition.onset[x]
- Removed condition.note.author[x] element
- Removed condition.note.time element
- Added recorder element extension - CareConnect-FamilyMemberHistoryRecorder-1-Extension 
- Added encounter element extension - http://hl7.org/fhir/StructureDefinition/encounter-associatedEncounter

Create new extensions:
- CareConnect-FamilyMemberHistoryRecorder-1-Extension

Commit:
ReferralRequest-1 and associated resources

ReferralRequest-1:
* Changed resource references to reference CareConnect profile versions.
* Add UBRN slice to identifier with 0..1 cardinality
* Made patient element mandatory
* Made requester element mandatory
* Made recipient element mandatory
* Added qualifier element extension to reason.coding -
CareConnect-CodingQualifier-1-Extension
* Changed supportingInformation element Definition

jonathanhomer commented:
ReferralRequest-1 and associated resources

ReferralRequest-1:
- Changed resource references to reference CareConnect profile versions.
- Add UBRN slice to identifier with 0..1 cardinality
- Made patient element mandatory
- Made requester element mandatory
- Made recipient element mandatory
- Added qualifier element extension to reason.coding - CareConnect-CodingQualifier-1-Extension
- Changed supportingInformation element Definition

Commit:
ReferralRequest-1 and associated resources

ReferralRequest-1:
* Changed resource references to reference CareConnect profile versions.
* Added qualifier element extension to code.coding -
CareConnect-CodingQualifier-1-Extension
* Removed note.author[x] element
* Removed note.time element
* Restricted orderer to Patient, Practitioner

jonathanhomer commented:
ReferralRequest-1 and associated resources

ReferralRequest-1:
- Changed resource references to reference CareConnect profile versions.
- Added qualifier element extension to code.coding - CareConnect-CodingQualifier-1-Extension
- Removed note.author[x] element
- Removed note.time element
- Restricted orderer to Patient, Practitioner

Commit:
CareConnect-SpecimenFasting-1-Extension

Observation-1:
* Changed resource references to reference CareConnect profile versions
* Changed code valueset binding strength to Preferred for valueset
reference http://hl7.org.uk/CareConnect-ObservationCode-1.valueset
* Added qualifier element extension to code.coding -
CareConnect-CodingQualifier-1-Extension
* Restricted subject to Patient
* Changed subject element Short description & Definition
* Changed effective[x] element Definition
* Made performer element mandatory
* Restricted value[x] to Quantity, string, Range and Ratio data types
* Changed component.code valueset binding strength to Preferred for
valueset reference
http://hl7.org.uk/CareConnect-ObservationCode-1.valueset
* Added qualifier element extension to component.code.coding -
CareConnect-CodingQualifier-1-Extension
* Restricted component.value[x] to Quantity, string, Range and Ratio
data types

Specimen-1:
* Changed resource references to reference CareConnect profile versions
* Restricted subject to Patient
* Added fasting element extension to collection -
CareConnect-SpecimenFasting-1-Extension

Create new valuesets:
* CareConnect-ObservationCode-1

Create new extensions:
* CareConnect-SpecimenFasting-1-Extension

jonathanhomer commented:
Observation-1, Specimen-1 and associated resources

Observation-1:
- Changed resource references to reference CareConnect profile versions
- Changed code valueset binding strength to Preferred for valueset reference http://hl7.org.uk/CareConnect-ObservationCode-1.valueset
- Added qualifier element extension to code.coding - CareConnect-CodingQualifier-1-Extension
- Restricted subject to Patient
- Changed subject element Short description & Definition
- Changed effective[x] element Definition
- Made performer element mandatory
- Restricted value[x] to Quantity, string, Range and Ratio data types
- Changed component.code valueset binding strength to Preferred for valueset reference http://hl7.org.uk/CareConnect-ObservationCode-1.valueset
- Added qualifier element extension to component.code.coding - CareConnect-CodingQualifier-1-Extension
- Restricted component.value[x] to Quantity, string, Range and Ratio data types

Specimen-1:
- Changed resource references to reference CareConnect profile versions
- Restricted subject to Patient
- Added fasting element extension to collection - CareConnect-SpecimenFasting-1-Extension

Create new valuesets:
- CareConnect-ObservationCode-1

Create new extensions:
- CareConnect-SpecimenFasting-1-Extension

Commit:
DiagnosticOrder-1, DiagnosticReport-1 and associated resources

DiagnosticOrder-1:
* Changed resource references to reference CareConnect profile versions
* Restricted subject to Patient
* Changed subject element Short description & Definition to reflect
Patient restriction
* Changed item.code valueset reference to
http://hl7.org.uk/CareConnect-DiagnosticOrderCode-1.valueset
* Removed note.author[x] element
* Removed note.time element

DiagnosticReport-1:
* Changed resource references to reference CareConnect profile versions
* Changed item.code valueset reference to
http://hl7.org.uk/CareConnect-DiagnosticReportCode-1.valueset
* Restricted subject to Patient
* Changed subject element Short description & Definition to reflect
Patient restriction
* Restricted request to DiagnosticOrder data type
* Added interpretation element extension -
CareConnect-DiagnosticReportInterpretation-1-Extension
* Added comment element extension -
CareConnect-DiagnosticReportComment-1-Extension

Create new valuesets:
* CareConnect-DiagnosticOrderCode-1
* CareConnect-DiagnosticReportCode-1

Create new extensions:
* CareConnect-DiagnosticReportInterpretation-1-Extension
* CareConnect-DiagnosticReportComment-1-Extension

jonathanhomer commented:
DiagnosticOrder-1, DiagnosticReport-1 and associated resources

DiagnosticOrder-1:
- Changed resource references to reference CareConnect profile versions
- Restricted subject to Patient
- Changed subject element Short description & Definition to reflect Patient restriction
- Changed item.code valueset reference to http://hl7.org.uk/CareConnect-DiagnosticOrderCode-1.valueset
- Removed note.author[x] element
- Removed note.time element

DiagnosticReport-1:
- Changed resource references to reference CareConnect profile versions
- Changed item.code valueset reference to http://hl7.org.uk/CareConnect-DiagnosticReportCode-1.valueset
- Restricted subject to Patient
- Changed subject element Short description & Definition to reflect Patient restriction
- Restricted request to DiagnosticOrder data type
- Added interpretation element extension - CareConnect-DiagnosticReportInterpretation-1-Extension
- Added comment element extension - CareConnect-DiagnosticReportComment-1-Extension

Create new valuesets:
- CareConnect-DiagnosticOrderCode-1
- CareConnect-DiagnosticReportCode-1

Create new extensions:
- CareConnect-DiagnosticReportInterpretation-1-Extension
- CareConnect-DiagnosticReportComment-1-Extension

Commit:
Convergence of Problem-Condition1 and associated resources with GP Connect

Problem—Condition-1: (New profile based on Condition)
* Changed resource references to reference CareConnect profile versions
* Removed used type, period & assigner from identifier element
* Made asserter element mandatory
* Made dateRecorded element mandatory
* Changed cardinality of code.coding to 1..*
* Add SNOMED CT slice to code.coding with fixed system value of
http://snomed.info/sct
* Added qualifier element extension to code.coding -
CareConnect-CodingQualifier-1-Extension
* Made category element mandatory
* Changed category valueset reference to
http://hl7.org.uk/CareConnect-ProblemConditionCategory-1.valueset
* Changed category valueset binding strength to Required
* Made category.coding.system element mandatory with a fixed value of
http://hl7.org.uk/CareConnect-ProblemConditionCategory-1.valueset
* Made category.coding.code element mandatory with a fixed value of
problem
* Made category.coding.display have a fixed value of Problem
* Changed clinicalStatus valueset binding strength to Required for
valueset reference http://hl7.org/fhir/ValueSet/condition-clinical
* Removed element severity.coding.version
* onset[x] data type restricted to dateTime e.g. onsetDateTime
* abatement[x] data type restricted to dateTime e.g. abatementDateTime
* Removed stage element
* Removed evidence element
* Removed bodySite element
* Added related element extension -
CareConnect-ProblemRelationship-1-Extension
* Added associated element extension -
CareConnect-ProblemAssociated-1-Extension

Create new valuesets:
* CareConnect-ProblemConditionCategory-1.valueset
* CareConnect-ProblemRelationshipType-1.valueset

Create new extensions:
* CareConnect-ProblemRelationship-1-Extension
* CareConnect-ProblemAssociated-1-Extension
* CareConnect-CodingQualifier-1-Extension

Commit:
Problem-Condition-1 Snapshot

Commit:
Convergence of Condition-1 and associated resources with GP Connect

Condition-1:
* Changed resource references to reference CareConnect profile versions
* Removed used type, period & assigner from identifier element
* Made asserter element mandatory
* Made dateRecorded element mandatory
* Changed cardinality of code.coding to 1..*
* Add SNOMED CT slice to code.coding with fixed system value of
http://snomed.info/sct
* Added qualifier element extension to code.coding -
CareConnect-CodingQualifier-1-Extension
* Removed category element
* Changed clinicalStatus valueset binding strength to Required for
valueset reference http://hl7.org/fhir/ValueSet/condition-clinical
* Removed element severity.coding.version
* onset[x] data type restricted to dateTime e.g. onsetDateTime
* abatement[x] data type restricted to dateTime e.g. abatementDateTime
* Removed stage element
* Removed evidence element
* Removed bodySite element
* Added problemEpisode element extension -
CareConnect-ConditionProblemEpisode-1-Extension

Create new valuesets:
* CareConnect-ConditionEpisodicity-1.valueset

Create new extensions:
* CareConnect-ConditionProblemEpisode-1-Extension
* CareConnect-CodingQualifier-1-Extension

Commit:
Resource Aggregation Mode added to Condition-1 & Problem-Condition-1 references

Commit:
Convergence of Observation-1 with GP Connect

Observation-1:
* Changed resource references to reference CareConnect profile versions
* Removed used type, period & assigner from identifier element
* Made identifier.value element mandatory
* Removed category element
* Add SNOMED CT slice to code.coding with fixed system value of
http://snomed.info/sct
* Added qualifier element extension to code.coding -
CareConnect-CodingQualifier-1-Extension
* Made subject element mandatory
* Restricted subject to Patient
* Set subject resource aggregation mode to Bundled
* Set encounter resource aggregation mode to Bundled
* effective[x] data type restricted to dateTime e.g. effectiveDateTime
* Removed issued element
* Made performer element mandatory
* Restricted performer to Patient or Practitioner
* Set performer resource aggregation mode to Referenced
* Restricted value[x] to Quantity, string, Range and Ratio data types
* Removed dataAbsentReason element
* Removed interpretation.coding.version &
interpretation.coding.userSelected elements
* Removed bodySite element
* Add SNOMED CT slice to method.coding with fixed system value of
http://snomed.info/sct
* Set specimen resource aggregation mode to Referenced
* Added fixed value of http://unitsofmeasure.org to
referenceRange.low.system
* Added fixed value of http://unitsofmeasure.org to
referenceRange.high.system
* Add SNOMED CT slice to referenceRange.meaning.coding with fixed system
value of http://snomed.info/sct
* Added fixed value of http://unitsofmeasure.org to
referenceRange.age.low.system
* Added fixed value of http://unitsofmeasure.org to
referenceRange.age.high.system
* Set related.target resource aggregation mode to Referenced
* Add SNOMED CT slice to component.code.coding with fixed system value
of http://snomed.info/sct
* Added qualifier element extension to component.code.coding -
CareConnect-CodingQualifier-1-Extension
* Restricted component.value[x] to Quantity, string, Range and Ratio
data types
* Removed component.dataAbsentReason element

Commit:
Add ReligiousAffiliation-1-Extension and add to Patient-1

Created the following:

ReligiousAffiliation-1 valueset
ReligiousAffiliation-1-Extension

Add the above to Patient-1

Commit:
Fix smart quotes in EthnicCategory-1

Commit:
Remove binding strength from Patient-1 ethnicCategory - this is on the extension itself

Commit:
Save snapshots for admin and appointment resources and extensions

Commit:
Convergence of MedicationOrder-1 with GP Connect

MedicationOrder-1:
* Changed resource references to reference CareConnect profile versions
* Removed used type, period & assigner from identifier element
* Made identifier.value element mandatory
* Made dateWritten element mandatory
* Made status element mandatory
* Add SNOMED CT slice to reasonEnded.coding with fixed system value of
http://snomed.info/sct
* Made patient element mandatory
* Set patient resource aggregation mode to Bundled
* Made prescriber element mandatory
* Set prescriber resource aggregation mode to Referenced
* Set encounter resource aggregation mode to Bundled
* Removed reason[x] element
* Restricted medication[x] to CodeableConcept
* Changed medicationCodeableConcept valueset binding strength to
Preferred for valueset reference
http://hl7.org.uk/CareConnect-ManufacturedMaterialSnCT-1.valueset
* Made dosageInstruction element mandatory
* Made dosageInstruction.text element mandatory
* Add SNOMED CT slice to dosageInstruction.additionalInstructions.coding
with fixed system value of http://snomed.info/sct
* Removed dosageInstruction.timing.event element
* Removed dosageInstruction.timing.repeat.bounds[x] element
* Removed dosageInstruction.timing.repeat.duration element
* Removed dosageInstruction.timing.repeat.durationMax element
* Removed dosageInstruction.timing.repeat.durationUnits element
* Removed dosageInstruction.timing.repeat.frequencyMax element
* Removed dosageInstruction.timing.repeat.periodMax element
* Removed dosageInstruction.timing.repeat.when element
* Removed dosageInstruction.timing.code element
* Removed dosageInstruction.asNeeded[x] element
* Removed dosageInstruction.site[x] element
* Removed dosageInstruction.route element
* Removed dosageInstruction.method element
* Removed dosageInstruction.dose[x] element
* Removed dosageInstruction.rate[x] element
* Removed dosageInstruction.maxDosePerPeriod element
* Made dispenseRequest element mandatory
* Removed dispenseRequest.medication[x] element
* Made dispenseRequest.validityPeriod.start element mandatory
* Made dispenseRequest.validityPeriod.end element mandatory
* Made dispenseRequest.quantity.value element mandatory
* Removed dispenseRequest.quantity.comparator element
* Made dispenseRequest.quantity.unit element mandatory
* Made dispenseRequest.quantity.system element mandatory
* Made dispenseRequest.quantity.code element mandatory
* Made dispenseRequest.expectedSupplyDuration.value element mandatory
* Removed dispenseRequest.expectedSupplyDuration.comparator element
* Made dispenseRequest.expectedSupplyDuration.unit element mandatory
* Made dispenseRequest.expectedSupplyDuration.system element mandatory
* Added fixed value of http://unitsofmeasure.org to
dispenseRequest.expectedSupplyDuration.system
* Made dispenseRequest.expectedSupplyDuration.code element mandatory
* Added quantityText element extension to dispenseRequest -
CareConnect-MedicationQuantityText-1-Extension
* Added reviewDate element extension to dispenseRequest -
CareConnect-MedicationRepeatReviewDate-1-Extension
* Removed substitution element
* Removed priorPrescription element

Create new valuesets:
* CareConnect-ManufacturedMaterialSnCT-1.valueset

Create new extensions:
* CareConnect-MedicationQuantityText-1-Extension
* CareConnect-MedicationRepeatReviewDate-1-Extension

Commit:
Convergence of AllergyIntolerance-1 and associated resources with GP Connect

AllergyIntolerance-1:
* Changed resource references to reference CareConnect profile versions
* Removed used type, period & assigner from identifier element
* Made identifier.value element mandatory
* Made recordedDate element mandatory
* Made recorder element mandatory
* Set recorder resource aggregation mode to Referenced
* Made patient element mandatory
* Set patient resource aggregation mode to Bundled
* Removed reporter element
* Changed comments on substance element to indicate that a SNOMED
AllergyCode may be sent where a direct substance mapping is not
available in the source system.
* Add SNOMED CT slice to substance.coding with fixed system value of
http://snomed.info/sct
* Removed criticality element
* Removed type element
* Removed category element
* Add SNOMED CT slice to reaction.substance.coding with fixed system
value of http://snomed.info/sct
* Removed reaction.certainty element
* Changed cardinality of reaction.manifestation to 1..*
* Add SNOMED CT slice to reaction.manifestation.coding with fixed system
value of http://snomed.info/sct
* Added qualifier element extension to reaction.manifestation.coding -
CareConnect-CodingQualifier-1-Extension
* Removed reaction.description element
* Removed reaction.exposureRoute element
* Removed reaction.note element
* Added encounter element extension -
http://hl7.org/fhir/StructureDefinition/encounter-associatedEncounter
* Added endDate element extension -
CareConnect-AllergyIntoleranceEndDate-1-Extension

Create new extensions:
* CareConnect-AllergyIntoleranceEndDate-1-Extension

Commit:
Save snapshot for resources and extensions (for Simplifier)

Commit:
Shorten URLs (for Simplifier) - Valuesets

Commit:
Shorten URLs (for Simplifier) - Extensions

Commit:
Shorten URLs (for Simplifier) - Resources

Commit:
Shorten URLs (for Simplifier) - Specific case for DiagnosticReportInterpretation-1-Ext

Commit:
Shorten URLs (for Simplifier) - Specific case for MedicationRepeatReviewDate-1-Ext

Commit:
Shorten URLs (for Simplifier) - Specific case for DiagnosticReportInterpretation-1-Ext

Commit:
Shorten URLs (for Simplifier) - Specific case for FamilyMemberHistoryRecorder-1-Ext

Commit:
Shorten URLs (for Simplifier) - Specific case for CareConnect-NhsNumberVerificationStatus-1-Extension

Commit:
Add short and definition fields to the root element of extensions

Commit:
Standardise valueset name field format

Commit:
Fix smart quotes on two valuesets

Commit:
Remove invalid valueset reference on CareConnect-Observation-1

Commit:
Reset structure definition maturity values

Commit:
Set resource maturity to 1 for converged resources

Commit:
MedicationOrder code binding strength set to Required

Commit:
MedicationOrder - Example values added for quanitity and expectedSupplyDuration

Commit:
MedicationOrder - quantityText extension moved to quantity element

Commit:
MedicationOrder - LastIssueDate extension added

Commit:
Fix convergence status of Observation and Condition

Commit:
AllergyIntolerance - removed note.author and note.time elements

Commit:
Change maturity of MedicationLastIssueDate-1-Ext

Commit:
Remove restrictions on meta.profile

Commit:
Condition-1 and associated resources

Commit:
Remove convergence marker from ConditionEpisodity extension and valueset as not referenced by convergence resource profiles

Commit:
Added SNOMED CT slice to stage.summary element

Commit:
Patient-1: Remove CHI and HC number identifier slices (as per DS)

Commit:
Patient-1: Remove other identifier slice and change to openAtEnd

Commit:
Patient-1:  Slice name and add a usual name slice.  Other name changes.

Commit:
Patient-1: Make gender and dateOfBirth mandatory.  Restrict deceased to date/time only.

Commit:
Patient-1: Require telecom.system and telecom.value and remove telecom.rank

Commit:
Patient-1:  Remove photo

Commit:
Patient-1:  Extend EthnicCategory-1 to cover full set of PDS codes

Commit:
Patient-1:  Add GPRegistrationDetails

Commit:
Update convergence status for Patient-1 and related extensions and valuesets

Commit:
Patient-1:  Add NHS communication details

Commit:
Patient-1:  Restrict multipleBirth to boolean only

Commit:
Patient-1:  Add OrganDonor extension

Commit:
Patient-1:  Add PersonRelationshipType valueset and change some Patient-1 cardinalities

Commit:
Patient-1:  Add residentialStatus and treatmentCategory

Commit:
Patient-1:  Add NhsNumberVerificationStatus and ResidentialStatus bindings

Commit:
Patient-1:  Add modeOfCommunication to NhsCommunication extension

Commit:
Condition Profile updated to include new valueset binding

Commit:
Fix oids, valueset references and update convergence set markers

Commit:
Add convergence marker to ConditionEpisode resources

Commit:
Remove Problem-Condition resources

Commit:
BodySite element added back into Condition and Observation resources

Commit:
AllergyIntolerance.reaction.manifestation.coding Qualifier removed

Commit:
CodingQualifier extension removed from convergance set

Commit:
RepeatInformation extension added to MedicationStatement

Commit:
Statement extension added to MedicationOrder

Commit:
MedicationStatement updated with inital feedback from DS

Commit:
Medication Statement added to Convergence Set
